core/block_processor.go

```diff
@@ -51,17 +51,12 @@ func (p *CommitProcessor) Process(tx *tx.Tx) error {
 	if err != nil {
 		panic(err)
 	}
-	err = executor.UpdateTrees()
```

Comment:
For committer, there is no need to `UpdateTrees`, but it is a necessary operation for witness. Because we are going to reuse the code, so I think we need think about how witness is going to reuse  the code before merging this PR.

Reply:
For witness processor, can call IntermediateRoot after each transaction to get the state root, and then clear the dirty objects to avoid redundant computation.

Reply:
Add a flag for IntermediateRoot, then we can reuse it.